sdk/servicebus/service-bus/samples/typescript/src/advanced/sessionManagerSample.ts

```diff
@@ -0,0 +1,155 @@
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+
+dotenv.config();
+
+const serviceBusConnectionString =
+  env["SERVICEBUS_CONNECTION_STRING"] || "<service bus connection string not in environment>";
+const queueName = env["QUEUE_NAME_WITH_SESSIONS"] || "<queue name not in environment>";
+const maxSessionsToProcessSimultaneously = 8;
+const sessionIdleTimeoutMs = 3 * 1000;
+const delayWhenNoSessionsAvailableMs = 5 * 1000;
+
+interface SessionMessageHandlers {
+  processInitialize(sessionId: string);
+  processMessage(msg: ReceivedMessageWithLock);
+  processError(err: Error, sessionId: string);
+  processClose(reason: "error" | "idle_timeout", sessionId: string);
+}
+
+function _createIdleTimer(
+  timeoutMs: number
+): { refresh(): void; expirationPromise: Promise<void> } {
+  let resolveExpirationPromise;
+
+  const expirationPromise = new Promise<void>((res, _rej) => {
+    resolveExpirationPromise = res;
+  });
+
+  let timer = setTimeout(resolveExpirationPromise, timeoutMs);
+
+  return {
+    refresh() {
+      clearTimeout(timer);
+      timer = setTimeout(resolveExpirationPromise, timeoutMs);
+    },
+    expirationPromise
+  };
+}
+
+async function _noSessionsAvailable(err: Error) {
+  // TODO: this error should be programmatically identifable.
+  return err.message.search(/Received an incorrect sessionId 'undefined'/) !== -1;
+}
+
+async function _processNextSession(
+  serviceBusClient: ServiceBusClient,
+  handlers: SessionMessageHandlers,
+  abortSignal: AbortSignalLike
+): Promise<void> {
+  const sessionReceiver = serviceBusClient.createSessionReceiver(queueName, "peekLock", {
+    autoRenewLockDurationInMs: sessionIdleTimeoutMs
+  });
+
+  try {
+    // TODO: I just picked a call that didn't grab a message here, just to ensure a session could
+    // actually be obtained. This goes away when we have `await createSessionReceiver()`.
+    await sessionReceiver.getState();
+  } catch (err) {
+    // TODO: when this happens do I still need to close the session receiver? (again, temporary concern)
+    if (_noSessionsAvailable(err)) {
+      console.log(`INFO: no available sessions, sleeping for ${delayWhenNoSessionsAvailableMs}`);
+      await delay(delayWhenNoSessionsAvailableMs);
+      return;
+    }
+
+    throw err;
+  }
+
+  await handlers.processInitialize(sessionReceiver.sessionId);
+
+  const idleTimer = _createIdleTimer(sessionIdleTimeoutMs);
+  let threwError = false;
+
+  sessionReceiver.subscribe(
+    {
+      async processMessage(msg) {
+        idleTimer.refresh();
+        await handlers.processMessage(msg);
+      },
+      async processError(err) {
+        threwError = true;
+        await handlers.processError(err, sessionReceiver.sessionId);
+      }
+    },
+    {
+      abortSignal
+    }
+  );
+
+  await idleTimer.expirationPromise;
+
+  try {
+    await sessionReceiver.close();
+  } catch (err) {
+    await handlers.processError(err, sessionReceiver.sessionId);
```

Comment:
I wouldnt put a try/catch around close() in a sample without first being clear on the kind of errors we would expect from it and the expected reactions from the customer.

Reply:
This is a good point. I believe we do have a larger conversation scheduled to talk about this so I don't mind having it here and then just adding doc comments to `close()` when we do finally do that.

Reply:
I do mind having it here and in no other samples.
If needed, you can log an issue to have the larger conversation and at the end of it update the samples as needed.

Reply:
Removed.